src/library/scala/collection/immutable/Range.scala

```diff
@@ -154,13 +154,13 @@ extends scala.collection.AbstractSeq[Int]
 
   @inline final override def foreach[@specialized(Unit) U](f: Int => U) {
     validateMaxLength()
-    val isCommonCase = (start != Int.MinValue || end != Int.MinValue)
+    val isCommonCase = start != Int.MinValue && start != Int.MaxValue && end != Int.MinValue && end != Int.MaxValue
     var i = start
     var count = 0
-    val terminal = terminalElement
```

Comment:
There was probably a reason for this to be stored in a local `val` before. Since it is now irrelevant to the fix, I think it should be kept as it was. Only the `val isCommonCase` line should be changed, basically.